### PR TITLE
💻 Make navbar responsive on mobile

### LINKED ISSUE #172: Fix navbar on mobile
Navbar looks janky on mobile. Wraps because there isn't enough space for all items. Fixable with a dropdown hamburger menu. Alternative solutions welcome.

### PR DESCRIPTION
**Pull request by evtyz**
### Before:
image
#### Tablet:
image
#### Phone: (This is animated but the screenshot is still)
image

## Purpose
The navbar was not responsive on devices with smaller screen width. This closes #172.

## Approach
Media queries to reduce margin and font size were used for tablets.
I created another div for the mobile dropdown and some CSS transitions for phones.

## Testing
Via visual inspection.

### CODE REVIEW

**Comment by evtyz** [comment]
Vercel preview is screwed up, idk why...

File: client/components/NavBar.tsx [hunk 1]
```diff
@@ -51,50 +66,136 @@ const LogoContainer = styled.div`
...
+  @media ${smallerThan(width.mobile)} {
+    flex-direction: column;
+    align-items: flex-end;
+  }
+`
+
+const NavBarContainer = styled.div`
+  position: fixed;
+  top: 0;
+  z-index: 1;
```
Comment (xujustinj) [inline client/components/NavBar.tsx:106]: See https://github.com/loolabs/waterpark/pull/177#issuecomment-881857536

```suggestion
  z-index: 8;
```

**Review by xujustinj — APPROVED** [review]
Very smooth animations, lgtm overall. See the z-index comment for a cheap bugfix we can get out of the way in this PR.

1 personal taste nit from me would be that the border around the hamburger feels unnecessary to me, we can leave it until a design review someday

File: client/components/NavBar.tsx (hunk 1 above)
Comment (evtyz) [reply client/components/NavBar.tsx]: Nice catch!

**Review by xujustinj — APPROVED** [review]

**Pull request merged**